lms/djangoapps/ccx/views.py

```diff
@@ -345,6 +363,14 @@ def set_grading_policy(request, course, ccx=None):
     override_field_for_ccx(
         ccx, course, 'grading_policy', json.loads(request.POST['policy']))
 
+    # using CCX object as sender here.
+    responses = SignalHandler.course_published.send(
+        sender=ccx,
+        course_key=CCXLocator.from_course_locator(course.id, unicode(ccx.id))
+    )
+    for rec, response in responses:
+        log.info('Signal fired when course is published. Receiver: %s. Response: %s', rec, response)
+
     url = reverse(
         'ccx_coach_dashboard',
         kwargs={'course_id': CCXLocator.from_course_locator(course.id, ccx.id)}
```

Comment:
Why does the signal line above need a `unicode(ccx.id)` but this `course_id` construction does not?